storage/innobase/row/row0merge.cc

```diff
@@ -4369,6 +4581,21 @@ row_merge_build_indexes(
 		DBUG_RETURN(DB_OUT_OF_MEMORY);
 	}
 
+	/* Get crypt data from tablespace if present. */
+	crypt_block = NULL;
+
+	/* If tablespace is encrypted, allocate additional buffer for
+	encryption/decryption. */
+	if (dict_table_is_encrypted(new_table)) {
+		crypt_block = static_cast<row_merge_block_t*>(
+			alloc.allocate_large(
+				3 * srv_sort_buf_size, &crypt_pfx));
```

Comment:
This might be ridiculously defensive but please consider (the upstream might rewrite the same and we wouldn't notice): extract srv_sort_buf_size sysvar max value to a define, add a static assert that 3 * max does not overflow os_offset_t

Reply:
Currently `sort_buffer_size` max value is 64 MiB, so it is very unlike to overflow. But if you insist, I will add static assert

Reply:
I don't insist :)